policyengine_api/utils/worker_logger.py

```diff
@@ -0,0 +1,74 @@
+from policyengine_api.utils.logger import Logger
+from rq import Worker, get_current_job
+from datetime import datetime
+
+
+class WorkerLogger(Logger):
+    """
+    Custom logger for worker processes
+    """
+
+    def __init__(
+        self,
+        dir: str = "logs",
+        name: str = "worker",
+        id: str = None,
+        log_to_cloud: bool = True,
+    ):
+        """
+        Initialize logger with automatic worker ID detection if none provided
+
+        All args optional
+        Args:
+            dir (str): Directory to store log files (defaults to "logs")
+            name (str): Name of the logger (defaults to "worker")
+            id (str): ID to append to log file name; if not provided, will fetch worker name
+            log_to_cloud (bool): Whether to log to Google Cloud Logging (defaults to True)
+            monitor_memory (bool): Whether to monitor memory usage (defaults to True)
+            memory_threshold (int): Memory usage threshold to trigger warnings (default: 75%)
+            memory_check_interval (int): How often to check memory in seconds (default: 5)
```

Comment:
**nitt, documentation** Arguments in documentation not in the constructor.  **suggestion** remove them